File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Hash.java
package com.onevgo.functions;

import cn.hutool.core.util.HexUtil;
import cn.hutool.crypto.digest.Digester;

import java.util.Arrays;

public class Hash {
    public static byte[] hash(String algo, byte[] data) {
        return new Digester(algo).digest(data);
    }

    public static String hash(String algo, byte[] data, boolean toLowerCase) {
        return HexUtil.encodeHexStr(hash(algo, data), toLowerCase);
    }

    public static void main(String[] args) {
        System.out.println(Arrays.toString(hash("ripemd160", "The quick brown fox jumped over the lazy dog.".getBytes())));
        System.out.println(hash("ripemd160", "The quick brown fox jumped over the lazy dog.".getBytes(), true));
    }
}

File: 01_Language/01_Functions/python/hash.py
# coding: utf-8

import hashlib


def hash(algo, data, raw_output=False):
    h = hashlib.new(algo)
    h.update(data)
    if raw_output:
        return h.digest()
    else:
        return h.hexdigest()


if __name__ == '__main__':
    print(hash('ripemd160', 'The quick brown fox jumped over the lazy dog.'.encode()))
